js/ui/handlers.js

```diff
@@ -132,13 +128,27 @@ function Handlers(map) {
             if (e.delta < 0 && scale !== 0) scale = 1 / scale;
 
             var fromScale = map.ease && isFinite(e.delta) ? map.ease.to : map.transform.scale,
+                targetZoom = map.transform.scaleZoom(fromScale * scale),
                 duration = !isFinite(e.delta) ? 800 : e.source === 'trackpad' ? 0 : 300;
 
-            map.zoomTo(map.transform.scaleZoom(fromScale * scale), {
+            map.zoomTo(targetZoom, {
                 duration: duration,
                 around: map.unproject(e.point)
             });
         })
+        .on('pinchstart', function() {
+            startScale = map.transform.scale;
+            startBearing = map.transform.bearing;
+        })
+        .on('pinch', function(e) {
+            var zoom = map.transform.scaleZoom(startScale * e.scale),
+                bearing = startBearing + e.bearing;
+
+            map.easeTo(null, zoom, bearing, {
```

Comment:
why not use `undefined` here instead of having add the null check on the other side?